plugin/pkg/scheduler/algorithm/priorities/priorities_test.go

```diff
@@ -929,3 +930,142 @@ func TestPrioritiesRegistered(t *testing.T) {
+							{
+							    "preferAvoidPods": [
+							        {
+							            "podSignature": {
+							                "podController": {
+							                    "kind": "ReplicationController",
+							                    "namespace": "default",
+							                    "name": "foo",
+							                    "apiVersion": "v1"
+							                }
+							            },
+							            "reason": "some reason",
+							            "message": "some message"
+							        }
+							    ]
+							}`,
+	}
+	annotations2 := map[string]string{
+		api.PreferAvoidPodsAnnotationKey: `
+							{
+							    "preferAvoidPods": [
+							        {
+							            "podSignature": {
+							                "podController": {
+							                    "kind": "ReplicationController",
+							                    "namespace": "default",
+							                    "name": "bar",
+							                    "apiVersion": "v1"
+							                }
+							            },
+							            "reason": "some reason",
+							            "message": "some message"
+							        }
+							    ]
+							}`,
+	}
+	tests := []struct {
+		pod          *api.Pod
+		rcs          []api.ReplicationController
+		nodes        []api.Node
+		expectedList schedulerapi.HostPriorityList
+		test         string
+	}{
+		{
+			pod: &api.Pod{ObjectMeta: api.ObjectMeta{Namespace: "default", Labels: label1}},
+			rcs: []api.ReplicationController{
+				{
+					ObjectMeta: api.ObjectMeta{
+						Namespace: "default",
+						Name:      "foo",
+					},
+					Spec: api.ReplicationControllerSpec{Selector: label1},
+				},
+			},
+			nodes: []api.Node{
+				{
+					ObjectMeta: api.ObjectMeta{Name: "machine1", Annotations: annotations1},
+				},
+				{
+					ObjectMeta: api.ObjectMeta{Name: "machine2", Annotations: annotations2},
+				},
+				{
+					ObjectMeta: api.ObjectMeta{Name: "machine3"},
+				},
+			},
+			expectedList: []schedulerapi.HostPriority{{"machine1", 0}, {"machine2", 10}, {"machine3", 10}},
+			test:         "pod should avoid a node, this node get lowest priority score",
+		},
+		{
+			pod: &api.Pod{ObjectMeta: api.ObjectMeta{Namespace: "default", Labels: label1}},
+			rcs: []api.ReplicationController{
+				{
+					TypeMeta: unversioned.TypeMeta{
+						APIVersion: "v1",
+						Kind:       "ReplicationController",
+					},
+					ObjectMeta: api.ObjectMeta{
+						Namespace: "default",
+						Name:      "foo",
+					},
+					Spec: api.ReplicationControllerSpec{Selector: label1},
+				},
+			},
+			nodes: []api.Node{
+				{
+					ObjectMeta: api.ObjectMeta{Name: "machine1", Annotations: annotations1},
+				},
+				{
+					ObjectMeta: api.ObjectMeta{Name: "machine2", Annotations: annotations1},
+				},
+			},
+			expectedList: []schedulerapi.HostPriority{{"machine1", 10}, {"machine2", 10}},
+			test:         "pod should avoid these nodes, but there's no other available nodes, all these nodes get highest priority score",
```

Comment:
this test (and the corresponding code) isn't necessary; see my comment in the priority function